### ISSUE TITLE
rapid response report crashes instructor dashboard

### ISSUE THREAD

**Issue by pdpinch** [mitodl/edx-platform#119]
### steps to reproduce

1. In studio, add Rapid Response to a problem
2. In lms, open the problem and close it (no need to generate any student data)
3. Back in studio, delete the problem. 
4. Back in lms, open the Instructor dashboard

### expected behavior

The instructor dashboard should open as usual. 

### actual behavior

The instructor dashboard crashes with a traceback similar to the one below and the browser request eventually times out. 

```
    2019-01-03 12:07:46,718 ERROR 11787 [django.request] exception.py:135 - Internal Server Error: /courses/course-v1:MITx+SCM.260r_4+2018_Fall/instructor
    Traceback (most recent call last):
      File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/exception.py", line 41, in inner
        response = get_response(request)
      File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 249, in _legacy_get_response
        response = self._get_response(request)
      File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 187, in _get_response
        response = self.process_exception_by_middleware(e, request)
      File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 185, in _get_response
        response = wrapped_callback(request, *callback_args, **callback_kwargs)
      File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/utils/decorators.py", line 185, in inner
        return func(*args, **kwargs)
      File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/newrelic/hooks/framework_django.py", line 544, in wrapper
        return wrapped(*args, **kwargs)
      File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/utils/decorators.py", line 149, in _wrapped_view
        response = view_func(request, *args, **kwargs)
      File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/views/decorators/cache.py", line 43, in _cache_controlled
        response = viewfunc(request, *args, **kw)
      File "/edx/app/edxapp/edx-platform/lms/djangoapps/instructor/views/instructor_dashboard.py", line 254, in instructor_dashboard_2
        return render_to_response('instructor/instructor_dashboard_2/instructor_dashboard_2.html', context)
      File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/shortcuts.py", line 169, in render_to_response
        return HttpResponse(render_to_string(template_name, dictionary, namespace, request), **kwargs)
      File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/shortcuts.py", line 159, in render_to_string
        return template.render(dictionary, request)
      File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/template.py", line 59, in render
        return self.mako_template.render_unicode(**context_dictionary)
      File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/template.py", line 454, in render_unicode
        as_unicode=True)
      File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/newrelic/hooks/template_mako.py", line 25, in __call__
        return self.__wrapped(template, *args, **kwargs)
      File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/runtime.py", line 829, in _render
        **_kwargs_for_callable(callable_, data))
      File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/runtime.py", line 864, in _render_context
        _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
      File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/runtime.py", line 890, in _exec_template
        callable_(context, *args, **kwargs)
      File "/tmp/mako_lms/84caac5afec1d5cec18c223971540436/mitx-theme/lms/templates/main.html.py", line 327, in render_body
        __M_writer(filters.decode.utf8(self.body()))
      File "/tmp/mako_lms/84caac5afec1d5cec18c223971540436/instructor/instructor_dashboard_2/instructor_dashboard_2.html.py", line 138, in render_body
        runtime._include_file(context, ( section_data['section_key'] ) + u'.html', _template_uri, section_data=section_data)
      File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/runtime.py", line 752, in _include_file
        callable_(ctx, **_kwargs_for_include(callable_, context._data, **kwargs))
      File "/tmp/mako_lms/84caac5afec1d5cec18c223971540436/instructor/instructor_dashboard_2/rapid_response.html.py", line 38, in render_body
        __M_writer(filters.html_escape(filters.decode.utf8(get_display_name_from_usage_key(run['problem_usage_key']))))
      File "/edx/app/edxapp/edx-platform/lms/djangoapps/instructor/views/tools.py", line 251, in get_display_name_from_usage_key
        return modulestore().get_item(key).display_name
      File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/modulestore/mixed.py", line 89, in inner
        retval = func(field_decorator=strip_key_collection, *args, **kwargs)
      File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/modulestore/mixed.py", line 255, in get_item
        return store.get_item(usage_key, depth, **kwargs)
      File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/modulestore/split_mongo/split_draft.py", line 280, in get_item
        return super(DraftVersioningModuleStore, self).get_item(usage_key, depth=depth, **kwargs)
      File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/modulestore/split_mongo/split.py", line 1222, in get_item
        items = self._load_items(course, [BlockKey.from_usage_key(usage_key)], depth, **kwargs)
      File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/modulestore/split_mongo/split.py", line 830, in _load_items
        return [runtime.load_item(block_key, course_entry, **kwargs) for block_key in block_keys]
      File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/modulestore/split_mongo/caching_descriptor_system.py", line 130, in _load_item
        block_data = self.get_module_data(block_key, course_key)
      File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/modulestore/split_mongo/caching_descriptor_system.py", line 158, in get_module_data
        raise ItemNotFoundError(block_key)
    ItemNotFoundError: BlockKey(type=u'problem', id=u'66ebdb175ae7408f937040ca921c1353')
```

I think we need to put some kind of protection around the `get_display_name_from_usage_key()` call in https://github.com/mitodl/edx-platform/blob/1be9ffdf048865bc336a6b18fd6cb417a4b0ce02/lms/templates/instructor/instructor_dashboard_2/rapid_response.html#L18

**Comment by Ayub-Khan** [comment]
@pdpinch I was able to reproduce the issue. I have created two prs with two solutions for the problem. [One](https://github.com/mitodl/edx-platform/pull/120) will show the data for removed blocks with name Deleted. [Second](https://github.com/mitodl/edx-platform/pull/121) one will not show the data for the removed blocks.

**Comment by pdpinch** [comment]
We merged #120. All this instructor dashboard code needs tests, and really should be implemented as a pluggable django app. But in the meantime, this issue is fixed.
